### ISSUE TITLE
Accessibility issues with the link wizard

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3451]
The link wizard that opens when you activate the 'internal link' button in the [edit view](https://grimm.haz.wiki/animals?do=edit) has multiple accessibility issues. Most of these affect screen reader users.

![](https://user-images.githubusercontent.com/108893/112769686-3b195f00-901a-11eb-936b-06d069fa755c.png)

* [ ] The text at the link wizard's top looks like a heading but isn't one.
That fails [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships).

* [ ] A link to a page and its description look like they belong together because they share the same background colour. But that is not programmatically determinable. (The list of pages could be a list.)
That fails [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships).

* [ ] The icons for the namespaces and pages have no clear alternative text. The colon (:) at the end of the namespace names indicate that they are namespaces, but those are not read out by screen readers (by default). That's made worse by the fact that links for namespaces do different things than the ones for pages. (See next item.)
That fails [1.1.1 Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content).

* [ ] The namespaces and pages in the link wizard are links but don't behave like links (i.e. they don't lead to other pages). They also behave differently from each other, which isn't programmatically determinable. Clicking on a page link puts that link into the textarea, clicking on a namespace link changes the view to list the pages and namespaces within that namespace. It's impossible to operate if you cannot distinguish one type of link from the other.
That fails [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value).

* [ ] When clicking on a namespace link or typing into the form field, the list of links gets updated. But neither that update nor what the form field does is programmatically determinable.
That fails [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value).

* [ ] When typing something into the form field which cannot be found, a status message is shown. ('Nothing was found.') But that status message popping up is not programmatically determinable.
That fails [4.1.3 Status Messages](https://www.w3.org/TR/WCAG21/#status-messages).

* [ ] The 'Link to:' input field doesn't have an accessible name (e.g. label).
That fails [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value).

## Other non-fail accessibility issues

These don't strictly fail WCAG but are still accessibility issues. Fixing them is recommended but optional for passing WCAG.

* [ ] The links could clarify their purpose either via their link text or via a heading.

* [ ] Typing into the 'link to' field changes the content below. But it's a change of content, not context. Pressing enter closes the dialog and inserts that word as a link. That is a change of context but somewhat anticipated. It could be made clearer what happens when to make sure it's not unexpected.

* [ ] After closing the link wizard (via the 'x' button or Esc), the focus goes to the textarea. It's more likely it will have been triggered from the textarea, so that would be expected in that case. But it might be a problem because unexpected if it was triggered from elsewhere.

* [ ] Not sure how reproducible this is and if it can be fixed. But when I tested the link wizard with NVDA in Firefox, after clicking on a namespace link it often read some of the text above the textarea ("Please edit the page only if you can improve it." etc). Although the focus stayed within the modal, which is good.

* [ ] When I tested with VoiceOver on iOS in Safari, sometimes following a link reloads the whole page. Same as before, not sure how reproducible this is and if it can be fixed. It might be down to a bug in VoiceOver.
